File: learn_android/code/JAVA/SharePreferencestest1/app/src/main/java/com/hkc/sharepreferencestest1/MainActivity.java
package com.hkc.sharepreferencestest1;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.util.Log;

public class MainActivity extends AppCompatActivity {
    int number;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        MyData myData = new MyData(getApplicationContext());

//        String a = getString(R.string.MY_DATA);
        myData.number = 1000;
        myData.save();
        int y = myData.load();
        Log.d("Mylog","onCreate " + y);


//        SharedPreferences shp = getPreferences(Context.MODE_PRIVATE);
//        SharedPreferences shp = getSharedPreferences("MyData",Context.MODE_PRIVATE);
//        SharedPreferences.Editor editor = shp.edit();
//        editor.putInt("NUMBER",100);
//        editor.apply();
//
//        number = shp.getInt("NUMBER",0);
//        String TAG = "MyLog";
//        Log.d(TAG,"on create :" + number);

    }
}

File: learn_android/code/BMIpractice1/app/src/main/java/com/hkc/bmipractice1/MainActivity.kt
package com.hkc.bmipractice1

import android.content.Intent
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Toast
import androidx.appcompat.app.AlertDialog
import com.hkc.bmipractice1.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
//        setContentView(R.layout.activity_main)

        binding = ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

/*        binding.bHelp.setOnClickListener {
            AlertDialog.Builder(this)
                .setMessage("體重(kg)/身高的平方(m)")
                .setTitle("BMI說明")
                .setPositiveButton("OK",null)
                .show()*/
//        }
        val app = getString(R.string.app_name)


    }

    fun bmi(view: View){
        val weight = binding.edWeight.text.toString().toFloat()
        val height = binding.edHeight.text.toString().toFloat()
        val bmi = weight / (height * height)
//        Log.d("BMI",bmi.toString())
        /*Toast.makeText(this,bmi.toString(),Toast.LENGTH_LONG).show()
        AlertDialog.Builder(this)
            .setMessage(bmi.toString())
            .setTitle("Your BMI")
            .setPositiveButton("OK",null)
            .setNeutralButton("Cancel",null)
            .show()*/
       /* val intent = Intent(this,ResultActivity::class.java)
        intent.putExtra("BMI_EXTRA",bmi)
        startActivity(intent)*/

        Intent(this,ResultActivity::class.java).apply{
            val bag = Bundle()
            bag.putFloat("BMI_EXTRA",bmi)
            bag.putString("TEST_EXTRA","Testing")
            putExtras(bag)
            startActivity(this)
    }


    }




}